devices/nand-imx6ull/dma.h

```diff
@@ -0,0 +1,82 @@
+/*
+ * Phoenix-RTOS
+ *
+ * Operating system loader
+ *
+ * i.MX 6ULL APBH DMA
+ *
+ * Copyright 2022 Phoenix Systems
+ * Author: Lukasz Kosinski
+ *
+ * This file is part of Phoenix-RTOS.
+ *
+ * %LICENSE%
+ */
+
+#ifndef _DMA_H_
+#define _DMA_H_
+
+#include <lib/lib.h>
+
+
+/* DMA descriptor flags */
+#define DMA_PIO(n)   (((n) & 0xf) << 12)
+#define DMA_HOT      (1 << 8)
+#define DMA_W4ENDCMD (1 << 7)
+#define DMA_DECSEMA  (1 << 6)
+#define DMA_W4READY  (1 << 5)
+#define DMA_NANDLOCK (1 << 4)
+#define DMA_IRQCOMP  (1 << 3)
+#define DMA_CHAIN    (1 << 2)
+#define DMA_SENSE    (3 << 0)
+#define DMA_READ     (2 << 0)
+#define DMA_WRITE    (1 << 0)
+#define DMA_NOXFER   (0 << 0)
+
+
+/* DMA descriptor */
+typedef struct {
+	u32 next;
+	u16 flags;
+	u16 size;
+	u32 addr;
+	u32 pio[];
+} __attribute__((packed)) dma_desc_t;
```

Comment:
Do we need "atomic" access to the fields of this struct? If we do, `__attribute__((packed))` will cross our plans

Reply:
No, we don't need atomic access to struct fields. I thinki that `__attribute__((packed))` is safe here.

Reply:
Ok, that was a problem in USB stack lately, but if you don't need to atomically write e.g. address, then it's ok indeed